joblib/parallel.py

```diff
@@ -775,17 +799,24 @@ def __call__(self, iterable):
         try:
             # Only set self._iterating to True if at least a batch
             # was dispatched. In particular this covers the edge
-            # case of Parallel used with an exhausted iterator.
+            # case of Parallel used with an exhausted iterator. If
+            # self._original_iterator is None, then this means either
+            # that pre_dispatch == "all", n_jobs == 1 or that the first batch
+            # was very quick and its callback already dispatched all the
```

Comment:
I removed it as redundant with `test_exhausted_iterator`.

I have tried to write a test for the specific case where the callback of the first predispatched jobs dispatches quickly the remaining job before entering the while loop but failed to trigger that case. But I am pretty it could happen.